### DISCUSSION TITLE
How to do out-of-source build?

### DISCUSSION THREAD

**Discussion by mloskot** [bfgroup/b2#145] [Q&A]
TL;TR: How to perform out-of-source build of Boost?

The [documentation about --build-dir says](https://www.bfgroup.xyz/b2/manual/main/index.html#bbv2.overview.invocation.options):

> The option is primarily useful when building from read-only media.

This is quite clear, but what about the bootstrap part?

Let's say I unpacked `boost_1_77_0.7z` into `C:\boost_1_77_0` and I want to keep the directory completely clean as it was read-only,

How to build Boost in such way that everything generated goes to separate `C:\boost_1_77_0_build` ?
How to mimic `cmake -S C:\boost_1_77_0 -B C:\boost_1_77_0_build`?

**Comment by grisumbras** [comment]
1. Build b2 separately
2. Create a project-config manually
3. Point b2 to that file via --project-config

Essentially, don't run bootstrap.sh

It still might not help you, if some script creates in-source files anyway.

**Reply by mloskot** [reply]
Thanks, but not quite what I was looking for.

I think I will stick to the pattern I currently use:
1. Unpack
2. Copy to build directory
3. Enter the build directory, boostrap and build in there.

My goal was to avoid the step 2.
